pkg/controllers/nodeclaim/disruption/drift_test.go

```diff
@@ -516,4 +520,51 @@ var _ = Describe("Drift", func() {
 			Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeDrifted)).To(BeNil())
 		})
 	})
+	Context("Reserved Capacity", func() {
+		var reservedOffering *cloudprovider.Offering
+		BeforeEach(func() {
+			reservedOffering = &cloudprovider.Offering{
+				Available: true,
+				Requirements: scheduling.NewLabelRequirements(map[string]string{
+					v1.CapacityTypeLabelKey:          v1.CapacityTypeReserved,
+					corev1.LabelTopologyZone:         "test-zone-1a",
+					cloudprovider.ReservationIDLabel: fmt.Sprintf("r-%s", it.Name),
+				}),
+				Price:               it.Offerings[0].Price / 1_000_000.0,
+				ReservationCapacity: 10,
+			}
+			it.Offerings = append(it.Offerings, reservedOffering)
+			it.Requirements.Get(v1.CapacityTypeLabelKey).Insert(v1.CapacityTypeReserved)
+
+			nodePool.Spec.Template.Spec.Requirements = append(nodePool.Spec.Template.Spec.Requirements, v1.NodeSelectorRequirementWithMinValues{
+				NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+					Key:      v1.CapacityTypeLabelKey,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{v1.CapacityTypeReserved},
+				},
+			})
+
+			for _, o := range []client.Object{nodeClaim, node} {
+				o.SetLabels(lo.Assign(o.GetLabels(), map[string]string{
+					v1.CapacityTypeLabelKey:          v1.CapacityTypeReserved,
+					cloudprovider.ReservationIDLabel: reservedOffering.ReservationID(),
+				}))
+			}
+
+			ctx = options.ToContext(ctx, test.Options(test.OptionsFields{FeatureGates: test.FeatureGates{ReservedCapacity: lo.ToPtr(true)}}))
+		})
+		// This is required to support cloudproviders dynamically updating the capacity type based on reservation expirations
+		It("should drift reserved nodeclaim if the capacity type label has been updated", func() {
```

Comment:
Is this not already covered by other drift testing? Kind of odd to me that we are specifically targeting this scenario when I would just expect it to be covered by generic drift requirement checking

Reply:
Good point, I don't think there's a reason we need to explicitly have this